### PR TITLE
Unify byz row and col errors.

### LINKED ISSUE #78: api: consolidate ErrByzantineCol/Row intos one ErrByzantineData
So that we handle row/cols branchlessly(without ifs to row/col). From what I observed, code that handles rows and cols can be unified into one code path within the lib itself and external users, e.g. celestia-node

### PR DESCRIPTION
**Pull request by adlerjohn**
Fixes #78

Unifies `ErrByzantineRow` and `ErrByzantineCol` into `ErrByzantineRowOrCol`. Provides two helper methods `IsRow()` and `IsCol()` since negating `isRow` to figure out if something is a column isn't immediately obvious.

### CODE REVIEW

**Review by vgonkivs — APPROVED** [review]

**Comment by adlerjohn** [comment]
I'm going to wait for @Wondertan to review for usage and @liamsi to review with write access.

File: extendeddatacrossword.go
Comment (Wondertan) [inline extendeddatacrossword.go:19]: Can we make a separate enum type for it? Something like: 
```go
type Axis int // or some other better generic name

const (
	Row Axis = iota
	Col
)

```

Comment (adlerjohn) [reply extendeddatacrossword.go:19]: Fun fact there's already an enum for that but it's unused :joy:. Probably from some older code that used it to branch.

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go:40]: Thinking about this a bit, is it actually more ergonomic to export `Row` and `Col` constants? Downstream users would have to do

```go
err.axis == rsmt2d.Row
```

which personally seems less ergonomic than `err.isRow()`.

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go]: Renamed in [d337c04](https://github.com/celestiaorg/rsmt2d/pull/80/commits/d337c042da11f9f9153620b346a4b52894f72b03)

File: extendeddatacrossword.go
Comment (Wondertan) [reply extendeddatacrossword.go]: True. I made enum public in the code snippet above

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go]: Fixed in [780a5bf](https://github.com/celestiaorg/rsmt2d/pull/80/commits/780a5bf8683982e369168079c758da86dbd072af)

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go]: Changed to use an enum (or whatever passes for an enum in Go...) in [780a5bf](https://github.com/celestiaorg/rsmt2d/pull/80/commits/780a5bf8683982e369168079c758da86dbd072af), but kept the helpers `isRow()` and `isCol()` as above.

**Review by Wondertan — COMMENTED** [review]
Can you please make axis and supporting consts public for users?

**Comment by adlerjohn** [comment]
Changed in [b1b3aff](https://github.com/celestiaorg/rsmt2d/pull/80/commits/b1b3aff0cd791bf868c90a69d682d958652e0fc6)

File: extendeddatacrossword.go
Comment (Wondertan) [inline extendeddatacrossword.go:9]: Capitalize.

File: extendeddatacrossword.go
Comment (Wondertan) [inline extendeddatacrossword.go:32]: Switch?

**Review by Wondertan — COMMENTED** [review]
Last nits

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go]: Fixed in [bd5044c](https://github.com/celestiaorg/rsmt2d/pull/80/commits/bd5044c996b04e39353775856cd8f5f6a158a182)

File: extendeddatacrossword.go
Comment (adlerjohn) [reply extendeddatacrossword.go]: Fixed in [778a54a](https://github.com/celestiaorg/rsmt2d/pull/80/commits/778a54a263a7612dfe51ab66a527b586ab774b86)

**Review by Wondertan — APPROVED** [review]
LGTM

**Review by liamsi — APPROVED** [review]

**Pull request merged**
